src/simulator/simulator.h

```diff
@@ -110,13 +110,23 @@ class Simulator : public ISimulator
 
 
 
-
     void initialize() override
     {
-        if (integrator_ != nullptr)
-            integrator_->initialize();
-        else
-            throw std::runtime_error("Error - Simulator has no integrator");
+        try
+        {
```

Comment:
This can leave the class in an invalid state, it should possibly rethrow.

example:

```
    try{
        throw std::runtime_error("VERY BAD");
    }
    catch (const std::runtime_error& e)
    {
        std::cout << e.what() << std::endl;
        std::rethrow_exception(std::current_exception());
        // or 
        //throw e;
    }
    catch (...)
    {
        std::cerr << "UNKNOWN EXCEPTION CAUGHT" << std::endl;
        exit(11); // exit on unknown error? should never happen, state is invalid if it does.
    }
```

Reply:
more a discussion point than requested change.

Reply:
I have no opinion about that, tell me if I need to do copy/paste your snippet

Reply:
here is the complete

```

    void initialize() override
    {
        try
        {
            if (integrator_ != nullptr)
                integrator_->initialize();
            else
                throw std::runtime_error("Error - Simulator has no integrator");
        }
        catch (const std::runtime_error& e)
        {
            std::cerr << "EXCEPTION CAUGHT: " << e.what() << std::endl;
            std::rethrow_exception(std::current_exception());
        }
        catch (...)
        {
            std::cerr << "UNKNOWN EXCEPTION CAUGHT" << std::endl;
            std::rethrow_exception(std::current_exception());
        }
    }
```

Reply:
done